File: chroma-league/src/test/java/com/bonepl/chromaleague/hud/parts/resource/YoneCloneBarTest.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.IntSteps;
import com.bonepl.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class YoneCloneBarTest extends AbstractResourceTest {

    @Test
    void testYoneCloneBar() {
        final IntSteps intSteps = new IntSteps(100, 0, 5);
        final YoneCloneBar yoneCloneBar = new YoneCloneBar();

        new AnimationTester()
                .withBeforeIterationAction(i -> mockResource(intSteps.nextInt(), 100))
                .testAnimation(yoneCloneBar, 20);
    }
}

File: test/hud/parts/resource/YoneCloneBar.test.ts
import {describe, test} from 'vitest';
import {IntSteps} from '../../../IntSteps.js';
import {AnimationTester} from '../../../AnimationTester.js';
import {YoneCloneBar} from '../../../../src/hud/parts/resource/YoneCloneBar.js';
import {mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for Yone clone resource bar on actual hardware. */
describe('YoneClone Integration', () => {
    setupResourceTest();

    test('testYoneCloneBar', async () => {
        const intSteps = new IntSteps(100, 0, 5);
        const yoneCloneBar = new YoneCloneBar();

        await new AnimationTester()
            .withBeforeIterationAction(() => mockResource(intSteps.nextInt(), 100))
            .testAnimation(yoneCloneBar, 20);
    }, 30000);
});
